analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -128,7 +128,12 @@ private static bool HasMutableUsagesInMethod(SemanticModel semanticModel, Variab
                 .Any(x => IsMutatingUse(semanticModel, x));
 
         static bool IsMethodLike(SyntaxNode arg) =>
-            arg is BaseMethodDeclarationSyntax or AccessorDeclarationSyntax or LambdaExpressionSyntax or GlobalStatementSyntax;
+            arg is BaseMethodDeclarationSyntax
+                or AccessorDeclarationSyntax
+                or LambdaExpressionSyntax
+                or AnonymousFunctionExpressionSyntax
+                or GlobalStatementSyntax
```

Comment:
I don't understand why `GlobalStatementSyntax` is considered a method.

Reply:
This is the boundary/scope of the variable: don't look any further up in the tree as any name you may find is either out of scope or shadowed so it doesn't make sense to look at it. GlobalStatementSyntax contains top-level statements so it is actually method-like.

Reply:
> GlobalStatementSyntax contains top-level statements so it is actually method-like.

It depends, in cases like:
```
using System;

Console.WriteLine("x");

Console.WriteLine("x");
```

the are 2 Global Statements but the variable scope is the same. We might get some hard to detect bugs depending on how we use this `IsMethodLike` method.

Reply:
Ah, it's inline. Nevermind.